app/collins/models/AssetLifecycle.scala

```diff
@@ -123,6 +125,13 @@ class AssetLifecycle(user: Option[User], tattler: Tattler) {
         opts.find(kv => restricted(kv._1)).map(kv =>
           return Left(new Exception("Attribute %s is restricted".format(kv._1))))
       }
+
+      opts.find(kv => Feature.encryptedTags.contains(kv._1)).map(kv =>
```

Comment:
map really seems like the wrong since you don't care about the results only the side effect.  perhaps foreach would more closely express what is happening

You might even put the writable and restricted tests into a single method since you use it in a few places